// Copyright 2015 gRPC authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

package workflowserver;

// The greeting service definition.
service Greeter {
  // test using
  rpc SayHello (HelloRequest) returns (HelloReply) {}
  // subscribe service
  rpc Subscribe (PubSubRequest) returns (PubSubReply) {}
  // publish service
  rpc Publish (PubSubRequest) returns (PubSubReply) {}

  rpc GetSubscribedNumber (SubNumRequest) returns (SubNumReply) {}

  rpc Notify (NotifyRequest) returns (NotifyReply) {}
}

// The request message containing the user's name.
message HelloRequest {
  string name = 1;
}

// The response message containing the greetings
message HelloReply {
  string message = 1;
}


message PubSubRequest {
  //string array
  repeated string pubsubmessage = 1;
  string clientid = 2;
}

message PubSubReply {
  string returnmessage = 1;
}


//determine how many clients subscribe this event 
//don't set the long name
message SubNumRequest {
  string subevent = 1;
}

message SubNumReply {
  int32 clientnumber = 1;
}

// workflow server send request back to eventNotify worker
message NotifyRequest {
  string clientid = 1;
}

//eventNotify worker return this reply message to workflow server
message NotifyReply {
  string returnmessage = 1;
}